list-doi-requests/src/test/java/no/unit/nva/doirequest/list/ListDoiRequestsHandlerTest.java

```diff
@@ -181,47 +195,130 @@ public void listDoiRequestsReturnsEmptyListForUserWhenUserHasNoValidRole()
+        createDoiRequests(publications);
+        final var doiRequestMessages = creteDoiRequestMessagesForPublications(publications);
+
+        URI commonPublisherId = publications.get(FIRST_ELEMENT).getPublisher().getId();
+        String commonOwner = publications.get(FIRST_ELEMENT).getOwner();
+        List<String> actualMessages = sendRequestAndReadMessages(commonOwner, commonPublisherId, CREATOR_ROLE);
+
+        String[] expectedMessages = extractExpectedMessageTexts(doiRequestMessages);
+        assertThat(actualMessages, containsInAnyOrder(expectedMessages));
+    }
+
+    @Test
+    public void listDoiRequestsForCuratorReturnsDtosWithMessagesIncludedWhenThereAreDoiRequestMessagesForTheDoiRequest()
+        throws ApiGatewayException, IOException {
+        var publications = createPublishedPublicationsOfSamePublisherButDifferentOwner();
+        createDoiRequests(publications);
+        final var doiRequestMessages = creteDoiRequestMessagesForPublications(publications);
+
+        URI commonPublisherId = publications.get(0).getPublisher().getId();
+        List<String> actualMessages = sendRequestAndReadMessages(SOME_CURATOR, commonPublisherId, CURATOR_ROLE);
+
+        String[] expectedMessages = extractExpectedMessageTexts(doiRequestMessages);
+        assertThat(actualMessages, containsInAnyOrder(expectedMessages));
+    }
+
+    public List<String> sendRequestAndReadMessages(String userIdentifier, URI commonPublisherId, String curatorRole)
+        throws IOException {
+        InputStream input = createRequest(commonPublisherId, userIdentifier, curatorRole);
+        handler.handleRequest(input, outputStream, context);
+        GatewayResponse<Publication[]> response = GatewayResponse.fromOutputStream(outputStream);
+
+        Publication[] doiRequestDtos = response.getBodyObject(Publication[].class);
+        return extractMessageTexts(doiRequestDtos);
+    }
+
+    private String[] extractExpectedMessageTexts(List<MessageDto> doiRequestMessages) {
+        List<String> texts = doiRequestMessages.stream()
+                                 .map(MessageDto::getText)
+                                 .collect(Collectors.toList());
+
+        String[] textArray = new String[texts.size()];
+        texts.toArray(textArray);
+        return textArray;
+    }
+
+    private List<String> extractMessageTexts(Publication[] doiRequestDtos) {
+        return Arrays.stream(doiRequestDtos)
+                   .map(Publication::getDoiRequest)
+                   .flatMap(d -> d.getMessages().stream())
+                   .map(DoiRequestMessage::getText)
+                   .collect(Collectors.toList());
+    }
+
+    private Environment mockEnvironment() {
+        Environment environment = mock(Environment.class);
+        when(environment.readEnv(ApiGatewayHandler.ALLOWED_ORIGIN_ENV)).thenReturn("*");
+        when(environment.readEnv(StorageModelConstants.HOST_ENV_VARIABLE_NAME)).thenReturn("localhost");
```

Comment:
Not very important, but should we put all strings in variables or use them as is? Might be redundant if its only used one place?